lib/auth/userloginstate/generator_test.go

```diff
@@ -257,14 +228,9 @@ func TestAccessLists(t *testing.T) {
 			members: append(newAccessListMembers(t, clock, "1", "user"), newAccessListMembers(t, clock, "2", "user")...),
 			roles:   []string{"role1"},
 			expected: newUserLoginState(t, "user",
-				[]string{
-					"role1",
-				},
-				trait.Traits{
-					"trait1": []string{"value1", "value2"},
-					"trait2": []string{"value3", "value4"},
-					"trait3": []string{"value5", "value6"},
-				}),
+				[]string{},
```

Comment:
Prefer `nil` to an empty slice, unless `newUserLoginState` actually cares about the difference.